tendrl/node_agent/manager/rpc.py

```diff
@@ -0,0 +1,111 @@
+import json
+import logging
+import re
+import traceback
+import uuid
+
+import etcd
+import gevent.event
+
+from tendrl.node_agent.config import TendrlConfig
+
+
+config = TendrlConfig()
+LOG = logging.getLogger(__name__)
+
+
+class EtcdRPC(object):
+
+    def __init__(self):
+        etcd_kwargs = {'port': int(config.get("bridge_common", "etcd_port")),
+                       'host': config.get("bridge_common", "etcd_connection")}
+
+        self.client = etcd.Client(**etcd_kwargs)
+        self.bridge_id = str(uuid.uuid4())
+
+    def _acceptor(self):
+        while True:
+            jobs = self.client.read("/api_job_queue")
+            for job in jobs.children:
+                raw_job = json.loads(job.value.decode('utf-8'))
+                # Pick up the "new" job that is not locked by any other bridge
+                if raw_job['status'] == "new":
```

Comment:
Another Option I see is having separate job queues for different jobs namely gluster_job_queue, ceph_job_queue, node_agent_job_queue from where gluster-bridge ceph-bridge and node-agent will pick the jobs respectively